NotificationServiceExtension/NotificationService.m

```diff
@@ -0,0 +1,75 @@
+//
+//  NotificationService.m
+//  NotificationServiceExtension
+//
+//  Created by Ivan Sein on 14.11.17.
+//  Copyright © 2017 struktur AG. All rights reserved.
+//
+
+#import "NotificationService.h"
+
+#import "UICKeyChainStore.h"
+#import "NCSettingsController.h"
+
+@interface NotificationService ()
+
+@property (nonatomic, strong) void (^contentHandler)(UNNotificationContent *contentToDeliver);
+@property (nonatomic, strong) UNMutableNotificationContent *bestAttemptContent;
+
+@end
+
+@implementation NotificationService
+
+- (void)didReceiveNotificationRequest:(UNNotificationRequest *)request withContentHandler:(void (^)(UNNotificationContent * _Nonnull))contentHandler {
+    self.contentHandler = contentHandler;
+    self.bestAttemptContent = [request.content mutableCopy];
+    
+    self.bestAttemptContent.title = @"Nextcloud notification 🔔";
+    self.bestAttemptContent.body = @"";
+    
+    NSString *message = [self.bestAttemptContent.userInfo objectForKey:@"subject"];
+    NSString *decryptedMessage = nil;
+    
+    @try {
+        UICKeyChainStore *keychain = [UICKeyChainStore keyChainStoreWithService:@"com.nextcloud.Talk"
+                                                                    accessGroup:@"group.com.nextcloud.Talk"];
+        decryptedMessage = [[NCSettingsController sharedInstance] decryptPushNotification:message
+                                                                     withDevicePrivateKey:[keychain dataForKey:kNCPNPrivateKey]];
+    } @catch (NSException *exception) {
+        NSLog(@"An error ocurred decrypting the message. %@", exception);
+    }
+    
+    if (decryptedMessage) {
+        NSData *data = [decryptedMessage dataUsingEncoding:NSUTF8StringEncoding];
+        id messageJSON = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+        NSString *app = [messageJSON objectForKey:@"app"];
+        NSString *type = [messageJSON objectForKey:@"type"];
+        NSString *subject = [messageJSON objectForKey:@"subject"];
+        
+        if ([app isEqualToString:@"spreed"]) {
+            self.bestAttemptContent.title = @"";
```

Comment:
Title empty?

Reply:
Yes :)

![img_0094](https://user-images.githubusercontent.com/4638605/33175664-bccb0c46-d05c-11e7-8d78-e8beebef0912.PNG)

Reply:
@jancborchardt What do you think about these alerts?

Reply:
OK, and what about chat? Why do we show that when we don't support chat on mobile devices? :)

Reply:
It will be supported at some point. (For now, we can show a message saying that it is not supported at the moment and provide and option to open web interface?)